Before Revert and all nukes changes, backported 6.2.1 code into it.

Kantanomo opened the pull request.

iSupersniper committed:
Merge pull request #288 from Kantanomo/development

Merging 0.6

iSupersniper committed:
Merge pull request #290 from Kantanomo/development

6.1.0 Changes

nukeulater committed:
[XLive] Removed unnecessary memory delete and XOverlappedGetExtendedError function 

Also handle XFriendsCreateErnumerator by returning ERROR_NO_MORE_FILES (meaning no xbox friends are to be retrieved)

nukeulater committed:
[3rdparty] Updated rapidjson (mainly fixed compiler warnings)

nukeulater committed:
[Utils] Added memory class

nukeulater committed:
[XNet/Sockets] Increase default recv/send buffers

This will benefit servers/clients on Windows 7 (maybe Server R2 as well)

nukeulater committed:
[Misc] Fixed frame pacing when using built-in frame limiter based off Sleep/std::this_thread::sleep_for function

Fixed some compiler warnings and include Memory.cpp in the project

nukeulater committed:
[XAchievements] Fix Overlapped parameter

nukeulater committed:
[Game] Updated players and game time globals

nukeulater committed:
[CustomResolutions] Fixed resolutions not showing up in windowed mode

nukeulater committed:
[xuser] Improved XUserReadProfileSettings stub

nukeulater committed:
[xlocator-serverlist] fixed serverlist buffer allocation size and server properties/string buffers memory leak

nukeulater committed:
[xlocator-serverlist] restore previous overlapped error results reporting, also fixed crash when closing the game while downloading the serverlist

nukeulater committed:
[utils] added onScreenDebugLog with formatting support

nukeulater committed:
[serverlist] log how many servers were found

nukeulater committed:
[misc] va_end here, oops

nukeulater committed:
[game] fixed fall damage fix lol

noticed it while playing, damage is server side u numb nuts

nukeulater committed:
[xlive] fixed serverlist not downloading in very rare cases

reset enumerator object handles to INVALID_HANDLE_VALUE when disposed

nukeulater committed:
[xnet] clear last registered session id key when it gets unregistered

nukeulater committed:
[misc] code cleanup and misc improvements

iSupersniper committed:
Merge pull request #293 from Kantanomo/development

6.1.1

nukeulater committed:
Merge remote-tracking branch 'pnill/development' into development-new

nukeulater committed:
[game] fixed live serverlist disappearing labels

got bored playing and decided to fix this

nukeulater committed:
[code] fixed some compiler warnings and improve/fix some macro usage

nukeulater committed:
[xnet] don't block joining if players join with the same account in another game instance

nukeulater committed:
[xnet] added some logging for connection bug

nukeulater committed:
[game] fix edge drop at higher tickrates

nukeulater committed:
[imgui] don't allow imgui to update the mouse cursor

should fix some lag when hovering over type-in settings menu (like FPS value, field of view)

nukeulater committed:
wtf is that code

storing values in random game memory then redirecting to that wtf man, just wtf

nukeulater committed:
make OnScreenDebugLog thread safe

nukeulater committed:
added some logging

nukeulater committed:
[game] fixed stutter on maps that override gamma settings and use bloom on NVIDIA/Intel cards

nukeulater committed:
fixed compile

nukeulater committed:
fix BadServer logging

nukeulater committed:
[game] improve projectile code

also added some research code from h3 i had laying around (disabled by default)

nukeulater committed:
[game] re-enable (or re-add) main loop throttle/frame limiter fixes while game is minimized

nukeulater committed:
[xnet] don't update time interaction when sending packets  

might cause the connection issue, though this is getting called in XNetGetConnectStatus as well

nukeulater committed:
[infection] don't force human weapons, the loadout can be set in the gametype/playlist

noticed someone in game chat asking about this, he actually made a good point

nukeulater committed:
explicitly overload datum index copy

nukeulater committed:
oops

nukeulater committed:
[xnet] code improvements and fixes, improved XSession implementation

nukeulater committed:
improve UncappedFps2 to not use byte code that overrides game function

nukeulater committed:
remove map downloading kick function, servers are now able to start without everyone having the map

nukeulater committed:
add a note

nukeulater committed:
[xnet] re-use connection spot rather than using the first one available

should allow to connect even if all the other connection spots are used, also add __FUCNTION__, why do i always fuck this up

nukeulater committed:
[logs] make creating logs easier and code modular

nukeulater committed:
fix onscreendebug init order

nukeulater committed:
create another log for critical network errors, that always gets created

nukeulater committed:
dont log this twice

nukeulater committed:
[xnet] create unique connection identification to determine if the connection attempt is new or not

nukeulater committed:
don't always log these, not that important

nukeulater committed:
[xnet] add connection initiator logic

nukeulater committed:
[xnet] allow peers that don't have the ports forwarded to connect to other peers that do have the ports open

nukeulater committed:
clean-up set_local_rank

had this code laying around

nukeulater committed:
[xsocket] send data to socket specified by the game, not by the socket

nukeulater committed:
[xnet] actual fix for peer to peer (even when 1 side has the ports open, and the other does not)

nukeulater committed:
slightly improve FPS in menus when using gaming mice that have the polling rate at 1000

not a full fix, in theory there is something wrong with the SetCursor() API in windows 10

nukeulater committed:
fix FPS drops in menus by limiting the max rate the cursor can update to 125hz (8 milliseconds) when receiving WM_SETCURSOR messages

don't spam SetCursor API

nukeulater committed:
optimize keyboard key states update by using GetKeyboardState() to prevent too many kernel calls

nukeulater committed:
adding missing code for compile

oops

nukeulater committed:
fix keyboard inputs for other game instances

nukeulater committed:
initialize critical_network_errors_log only if network_log is disabled

nukeulater committed:
fix deadlock when using "\n" in addDebugText

nukeulater committed:
call postConfig() actually after reading configuration file

Kantanomo committed:
Merge branch 'pr/13' into Actual-Before-Revert

Kantanomo committed:
Some fixes

Stuff I had to fix last time and accidently accepted the change from previous rather than the incoming changes during hte PR.

Kantanomo committed:
Migrating code from development-old

nukeulater committed:
fix ip_lan and ip_wan getting overridden if config is empty or invalid

nukeulater committed:
fix critical network log when all logs are enabled

fucking macros men

nukeulater committed:
[network] add automatic LAN address lookup, that can be overridden by config in case the retrieved address doesn't work

NukeULater commented:
Details:

Fixed built in frame limiters and stuttering by setting the global timer resolution.
More info here https://randomascii.wordpress.com/2020/10/04/windows-timer-resolution-the-great-rule-change/
Applies to Windows 10 v2004 and above (maybe bellow as well, if other applications set the global resolution timer)

Fixed serverlist buffer size calculation and a memory leak when handling strings.

Fixed LIVE serverlist from dropping UI labels when hovering over player populated servers.

XNet code improvements and implemented some other XLive APIs better.

Fixed performance when using high polling rate mice, causing the game to drop framerate when moving around when the cursor is visible, by limiting how many times the cursor indicator can be set to 125hz maximum on each WM_SETCURSOR window procedure message.

Improved performance of keyboard state polling by using just one call to GetKeyboardState instead of calling GetAsyncKeyState 256 times.

Fixed a game bug when tickrate is increased from 30.

improved the logging code, now logs can always be created at each point, the code is not hardcoded by H2Configs anymore, also re-enabled OnScreenDebug to log everytime even when logs are disabled by default, and added always enabled Critical Network log to log any critical network issues.

And misc code cleanup

nukeulater committed:
fix some logging macros

nukeulater committed:
fix typo

Kantanomo committed:
Adding New Render Mode

Moved nukes UncappedFPS method in exerpimental_rendering_mode's

disabled code added to paramters_update.

Kantanomo committed:
Refactor of Default packet overrides.

Refactored code where we are overriding default game packets.

Started overriding the syncronous simulation packets, nothing is active right now.

Kantanomo committed:
Merge branch 'pr/14' into development-new

Kantanomo committed:
No need to allocate memory, fixing a C17 compiler issue.

Kantanomo committed:
Switching everything to use the Memory namespace instead of h2mod.

Kantanomo committed:
Completely removing h2mod memory calls.

Kantanomo committed:
Thanks Himanshu.
